### PR TITLE
Show spinners after the prompts are responded and close the prompts on `ctrl + c` in the cli

### PR DESCRIPTION
**Pull request by JayaKrishnaNamburu**
## Description

This PR
- Allows the prompts to be cancelable at any time.
- Prompts the user if the dependencies can be installed, when the project is being connected for the first time.
- Shows the spinner once all the prompts are responded. Which was previously clearing the prompts to be displayed.

## Code Review

- [ ] hi @istarkov , I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review
- [x] made a self-review

### CODE REVIEW

File: packages/cli/src/commands/init-flow.ts [hunk 1]
```diff
@@ -13,46 +13,71 @@ export const initFlow = async (
...
   if (isProjectConfigured === false) {
-    prompsList.push(
+    const { createFolder } = await prompts([
       {
         type: "confirm",
-        name: "folder",
+        name: "createFolder",
         message: "Do you want to create a folder",
         initial: true,
+        onState: (state) => {
```
Comment (istarkov) [inline packages/cli/src/commands/init-flow.ts:24]: Mb wrap onState logic into own propmts wrapper
```
import prompts from './promts';// <-- our wrapper
await prompts({ type, name, message, initial });
```

File: packages/cli/src/commands/init-flow.ts [hunk 2]
```diff
@@ -13,46 +13,71 @@ export const initFlow = async (
...
+    }
+    await link({ link: projectLink });
   }
 
-  /*
-    If the project is already set up, we can sync and build it.
-  */
   await sync();
   await build(options);
+  const spinner = ora().start();
```
[the remark this answers is not in the document]
Comment (JayaKrishnaNamburu) [reply packages/cli/src/commands/init-flow.ts]: They have their own spinner states inside, when the commands are ran independently.

File: packages/cli/src/commands/init-flow.ts [hunk 3]
```diff
@@ -13,58 +13,90 @@ export const initFlow = async (
   options: StrictYargsOptionsToInterface<typeof buildOptions>
 ) => {
   const isProjectConfigured = await isFileExists(".webstudio/config.json");
-  const prompsList: PromptObject[] = [];
-  const spinner = ora().start();
+  let shouldInstallDeps = false;
 
   if (isProjectConfigured === false) {
-    prompsList.push(
+    const { createFolder } = await prompts([
```
Comment (TrySound) [inline packages/cli/src/commands/init-flow.ts:19]: shouldCreateFolder is better name, createFolder implies function

File: packages/cli/src/commands/init-flow.ts [hunk 4]
```diff
@@ -7,64 +6,68 @@ import ora from "ora";
...
-        name: "folder",
-        message: "Do you want to create a folder",
-        initial: true,
-      },
-      {
-        type: (prev) => (prev === true ? "text" : null),
+    const { shouldCreateFolder } = await prompt({
+      type: "confirm",
+      name: "shouldCreateFolder",
+      message: "Do you want to create a folder",
```
Comment (kof) [inline packages/cli/src/commands/init-flow.ts:22]: "Would you like to create a folder?"

File: packages/cli/src/commands/init-flow.ts [hunk 5]
```diff
@@ -7,64 +6,68 @@ import ora from "ora";
...
         message: "Enter a project name",
-      },
-      {
-        type: (prev) => (prev === undefined ? null : "text"),
-        name: "projectLink",
-        message: "Enter a project link",
+      });
+
+      if (folderName === undefined) {
+        throw new Error("Folder name is required");
```
Comment (kof) [inline packages/cli/src/commands/init-flow.ts:34]: why? if they don't enter it, it goes to "./" like we discussed

File: packages/cli/src/commands/init-flow.ts [hunk 6]
```diff
@@ -7,64 +6,68 @@ import ora from "ora";
...
-    await link({ link: response.projectLink });
+    if (projectLink === undefined) {
+      throw new Error(`Project Link is required`);
+    }
+    await link({ link: projectLink });
+
+    const { installDeps } = await prompt({
+      type: "confirm",
+      name: "installDeps",
+      message: "Do you want to install dependencies",
```
Comment (kof) [inline packages/cli/src/commands/init-flow.ts:54]: Would you like to install dependencies?

Comment (kof) [reply packages/cli/src/commands/init-flow.ts:54]: We need to treat the cli messaging like we are dealing with a customer in a UI

Comment (kof) [reply packages/cli/src/commands/init-flow.ts:54]: Please check all messages, run them through grammarly or notion ai for correctness

File: packages/cli/src/commands/init-flow.ts (hunk 5 above)
Comment (JayaKrishnaNamburu) [reply packages/cli/src/commands/init-flow.ts:34]: It does go, this is if they selected the option to create a folder and still don't enter 👍

**Pull request merged**